### ISSUE TITLE
Pipeline doesn't compile

### ISSUE THREAD

**Issue by romeokienzler** [elyra-ai/elyra#3020]
**Describe the issue**
Pipeline doesn't compile

**To Reproduce**
1. Clone this [repo](https://github.com/claimed-framework/component-library/blob/f8f600bef27cd9347f921c8d64e29a9e211c2991/pipelines/gharchive_kfp.pipeline)
2. Run [this](https://github.com/claimed-framework/component-library/blob/f8f600bef27cd9347f921c8d64e29a9e211c2991/pipelines/gharchive_kfp.pipeline) pipeline at the the specific commit

**Screenshots or log output**

Traceback (most recent call last):
  File "/home/romeokienzler/venvs/elyra/lib64/python3.10/site-packages/elyra/pipeline/kfp/processor_kfp.py", line 287, in process
    pipeline_dsl = self._generate_pipeline_dsl(
  File "/home/romeokienzler/venvs/elyra/lib64/python3.10/site-packages/elyra/pipeline/kfp/processor_kfp.py", line 546, in _generate_pipeline_dsl
    workflow_tasks = self._generate_workflow_tasks(
  File "/home/romeokienzler/venvs/elyra/lib64/python3.10/site-packages/elyra/pipeline/kfp/processor_kfp.py", line 923, in _generate_workflow_tasks
    property_value = f.read() if os.path.getsize(absolute_path) else None
  File "/usr/lib64/python3.10/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/romeokienzler/venvs/elyra/lib64/python3.10/site-packages/tornado/web.py", line 1713, in _execute
    result = await result
  File "/home/romeokienzler/venvs/elyra/lib64/python3.10/site-packages/elyra/pipeline/handlers.py", line 156, in post
    response = await PipelineProcessorManager.instance().process(pipeline)
  File "/home/romeokienzler/venvs/elyra/lib64/python3.10/site-packages/elyra/pipeline/processor.py", line 166, in process
    res = await asyncio.get_event_loop().run_in_executor(None, processor.process, pipeline)
  File "/usr/lib64/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/romeokienzler/venvs/elyra/lib64/python3.10/site-packages/elyra/pipeline/kfp/processor_kfp.py", line 304, in process
    raise RuntimeError(
RuntimeError: Error compiling pipeline 'gharchive_kfp' with engine 'argo'.

**Expected behavior**

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: [e.g. 1.5.3]
 - Operating system: [e.g. macos, linux]
 - Installation source: [e.g. PyPI, conda, from source, official container image, custom container image]
 - Deployment type: [e.g. local installation, Docker, Kubernetes, Kubeflow [notebook server] , Open Data Hub]

**Pipeline runtime environment**
If the issue is related to pipeline execution, identify the environment where the pipeline is executed
 - Local execution
 - Kubeflow Pipelines (provide version number, whether multi-user auth enabled)
 - Apache Airflow (provide version number)
 
**Runtime configuration settings**
If the issue is related to pipeline execution, document the runtime configuration settings from the Elyra UI, omitting confidential information.

**Comment by romeokienzler** [comment]
Just noticed this happens when a local file is selected as input (rather than a value or upstream output)

**Comment by ptitzler** [comment]
After further investigation (in the context of Kubeflow Pipelines as a runtime environment) I am removing the milestone and adding the `needs discussion` label. 

TL;DR:  Should the VPE support binary files as input for property values

In this scenario a binary file was selected as input for a (custom) component _property value_. When a user  exports a pipeline or submits it, Elyra generates YAML artifacts, which KFP processes. These artifacts contain the property value, not the file name. The problem is that binary data can only be included in YAML if it is base64 encoded. If we were to encode the input using base64 then KFP would pass it to the component as-is [1], and the component would have to know that it needs to decode it make it usable.

[1] The data is passed as a parameter to the component, i.e `program --my-input=<DATA>`

In general I think binary data should be exchanged between custom components as files:
 - either by modeling an explicit output/input property relationship (handled natively by KFP, and upon invocation the component would have access to the data via a file handle)
 - or by 'uploading'/'downloading' it to/from shared storage (object storage, or shared volumes, for example)
